sunpy/net/jsoc/jsoc.py

```diff
@@ -483,189 +639,166 @@ def get_request(self, requestIDs, path=None, overwrite=False, progress=True,
-        object in TAI frame. Alternatively convert a astropy time object to TAI
-
-        Parameters
-        ----------
-        time: six.string_types or datetime or astropy.time
-            Input time
-
-        Returns
-        -------
-        datetime, in TAI
+        Take the query arguments and build a record string.
         """
-        # Convert from any input (in UTC) to TAI
-        if isinstance(time, six.string_types):
-            time = parse_time(time)
-
-        time = astropy.time.Time(time, scale='utc')
-        time = time.tai  # Change the scale to TAI
-
-        return time.datetime
-
-    def _make_recordset(self, start_time, end_time, series, wavelength='',
-                        segment='', **kwargs):
-        # Build the dataset string
-        # Extract and format Wavelength
-        if wavelength:
-            if not series.startswith('aia'):
-                raise TypeError("This series does not support the wavelength attribute.")
-            else:
-                if isinstance(wavelength, list):
-                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
-                    wavelength = str(wavelength)
-                else:
-                    wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
 
         # Extract and format segment
-        if segment != '':
+        if segment:
+            if isinstance(segment, list):
+                segment = str(segment)[1:-1].replace(' ', '').replace("'", '')
             segment = '{{{segment}}}'.format(segment=segment)
 
+        # Extract and format sample
         sample = kwargs.get('sample', '')
         if sample:
             sample = '@{}s'.format(sample)
 
-        dataset = '{series}[{start}-{end}{sample}]{wavelength}{segment}'.format(
-                   series=series, start=start_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
-                   end=end_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
-                   sample=sample,
-                   wavelength=wavelength, segment=segment)
+        # Populate primekeys dict with Time and Wavelength values
+        if start_time and end_time:
+            if not any(x in PKEY_LIST_TIME for x in primekey):
+                timestr = '{start}-{end}{sample}'.format(
+                        start=start_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
+                        end=end_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
+                        sample=sample)
+            else:
+                error_message = "Time attribute has been passed both as a Time()"\
+                                " and PrimeKey(). Please provide any one of them"\
+                                " or separate them by OR operator."
+                raise ValueError(error_message)
 
-        return dataset
+        else:
+            for _ in PKEY_LIST_TIME:
+                timestr = '{0}'.format(primekey.pop(_, ''))
```

Comment:
What is the purpose of this loop? Are you trying to find the first matching key in the `primekey` dict with a string in `PKEY_LIST_TIME`? If so some set trickery would probably be better.

```python
 ❯ keys = ['Time', 'wibble']

 ❯ primekeys = {'bob': 1, 'Time': 2}

 ❯ set(primekeys.keys()) & set(keys)
{'Time'}
```
Also this makes it easy to check for multiple matches.

_This is super easy in Python 3 as `dict.keys()` actually returns a set, so I would recommend you make the  `PKEY_LIST_TIME` a set when it is defined, and then when we go Python 3 only it becomes a simple `primekey.keys() & PKEY_SET_TIME`_